lib/Controller/MessagesController.php

```diff
@@ -525,6 +526,43 @@ public function downloadAttachment(int $id,
 		);
 	}
 
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 * @TrapError
+	 *
+	 * @param int $id the message id
+	 * @param string $attachmentId
+	 *
+	 * @return ZipResponse|JSONResponse
+	 *
+	 * @throws ClientException
+	 * @throws ServiceException
+	 * @throws DoesNotExistException
+	 */
+	public function downloadAttachments(int $id): Response {
+		try {
+			$message = $this->mailManager->getMessage($this->currentUserId, $id);
+			$mailbox = $this->mailManager->getMailbox($this->currentUserId, $message->getMailboxId());
+			$account = $this->accountService->find($this->currentUserId, $mailbox->getAccountId());
+		} catch (DoesNotExistException $e) {
+			return new JSONResponse([], Http::STATUS_FORBIDDEN);
+		}
+
+		$attachments = $this->mailManager->getMailAttachments($account, $mailbox, $message);
+		$zip = new ZipResponse($this->request, 'attachments');
+
+		foreach ($attachments as $attachment) {
+			$fileName = $attachment['name'];
+			$fh = fopen("php://temp", 'r+');
+			fputs($fh, $attachment['content']);
+			$size = (int)$attachment['size'];
+			rewind($fh);
+			$zip->addResource($fh, $fileName, $size);
```

Comment:
```suggestion
			$fileName = $attachment['name'];
			$fh = fopen("php://temp", 'r+');
			fputs($fh, $attachment['content']);
			$size = (int)$attachment['size'];
			rewind($fh);
			$zip->addResource($fh, $fileName, $size);
			fclose($fh);
```

Does php automatically close files that go out of scope? If not it should be closed manually here.

Reply:
After having a deeper look I think it's not quite right to close the stream here already. The reason is that `addResource` won't consume the stream right away but only when the response data is rendered and streamed. So if we close this here, the mem stream won't be available for the zip data later on.

Reply:
My bad. Thanks for digging the code.

Reply:
No worries. I appreciate the thorough review :pray: